src/metabase/models/setting.clj

```diff
@@ -1342,3 +1343,42 @@
            (map (fn [[setting-name]]
                   [setting-name (get setting-name)])))
      @registered-settings)))
+
+(defn- set-via-env? [setting]
+  (contains? env/env (setting-env-map-name setting)))
+
+(defmulti ^:private validate-setting-format :type)
```

Comment:
Even though this is private, I'd suggest adding a docstring so we know what contract each implementation needs to fulfill. Multimethods can easily become inconsistent otherwise. At the very least, you should document "Returns nil if the value passes validation"

Reply:
Agreed, I was just putting off writing it until I added the next type. Writing is hard, but fine I'll do it now 😄